scripts/run_changed.js

```diff
@@ -17,19 +17,55 @@
 
 const { resolve } = require('path');
 const { spawn } = require('child-process-promise');
+const chalk = require('chalk');
 const simpleGit = require('simple-git/promise');
+
 const root = resolve(__dirname, '..');
 const git = simpleGit(root);
 
 /**
- * Runs tests on packages changed (in comparison to origin/master...HEAD)
+ * Changes to these files warrant running all tests.
+ */
+const fullTestTriggerFiles = [
+  // Global dependency changes.
+  'package.json',
+  'yarn.lock',
+  // Test/compile/lint configs.
+  'config/karma.base.js',
+  'config/.eslintrc.js',
+  'config/mocha.browser.opts',
+  'config/mocha.node.opts',
+  'config/tsconfig.base.json',
+  'config/webpack.test.js',
+  'config/firestore.rules',
+  'config/database.rules.json'
+];
+
+/**
+ * Always run tests in these paths.
  */
+const alwaysRunTestPaths = [
+  // These tests are very fast.
+  'integration/browserify',
+  'integration/firebase-typings',
+  'integration/typescript',
+  'integration/webpack'
+];
 
-async function runTestsOnChangedPackages() {
+/**
+ * Identify modified packages that require tests.
+ */
+async function getChangedPackages() {
   const diff = await git.diff(['--name-only', 'origin/master...HEAD']);
   const changedFiles = diff.split('\n');
   const changedPackages = {};
   for (const filename of changedFiles) {
+    if (fullTestTriggerFiles.includes(filename)) {
+      console.log(
+        chalk`{blue Running all tests because ${filename} was modified.}`
```

Comment:
I'm not familiar with this syntax. Hope it's correct.

Reply:
Seems to work when run.